src/arrayMethodSort.js

```diff
@@ -4,9 +4,42 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function(compareFunction = compareStrings) {
     // write code here
+    let counter;
+
+    do {
+      counter = 0;
```

Comment:
since this variable is not really counting (it's value is either 0 or 1) I would suggest changing the name to something like isSorting, and using boolean values (true / false) instead of 0 / 1